adapters/rest-vertx/src/main/java/org/eclipse/hono/adapter/rest/credentials/BasicAuthProvider.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2017 Bosch Software Innovations GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial creation
+ */
+
+package org.eclipse.hono.adapter.rest.credentials;
+
+import io.vertx.core.*;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.auth.AuthProvider;
+import io.vertx.ext.auth.User;
+import org.eclipse.hono.adapter.rest.VertxBasedRestProtocolAdapter;
+
+/**
+ * Helper class implementing {@link AuthProvider}. The Vertx components handle the header extraction and return the
+ * appropriate error codes if the user authorization fails.
+ */
+public class BasicAuthProvider implements AuthProvider {
+
+    private VertxBasedRestProtocolAdapter adapter;
+
+    public BasicAuthProvider(VertxBasedRestProtocolAdapter adapter) {
```

Comment:
params are final
is there a reason why this is public when we also have a static factory method (`create`)?